src/main/twirl/uk/gov/hmrc/hmrcfrontend/views/components/HmrcCharacterCount.scala.html

```diff
@@ -14,55 +14,79 @@
  * limitations under the License.
  *@
 
-@import uk.gov.hmrc.govukfrontend.views.html.components.GovukTextarea
-@import uk.gov.hmrc.govukfrontend.views.viewmodels.textarea.Textarea
-@import uk.gov.hmrc.govukfrontend.views.viewmodels.content.Text
-@import uk.gov.hmrc.govukfrontend.views.html.components.GovukHint
-@import uk.gov.hmrc.hmrcfrontend.views.viewmodels.language.{Cy, En}
-@import uk.gov.hmrc.govukfrontend.views.viewmodels.hint.Hint
+@import uk.gov.hmrc.govukfrontend.views.html.components.GovukCharacterCount
+@import uk.gov.hmrc.govukfrontend.views.Aliases.{CharacterCount => GovukFrontendCharacterCount}
 
-@this(govukTextarea: GovukTextarea, govukHint: GovukHint)
+@this(govukCharacterCount: GovukCharacterCount)
 @(params: CharacterCount)(implicit messages: Messages)
 @import params._
 
 @require(name.nonEmpty && id.nonEmpty, "name and id should be non-empty")
 @require(maxLength.isDefined || maxWords.isDefined, "maxLength or maxWords should be provided")
 
-@attrs = {
-  @params.maxLength.filter(_ > 0).map {value => data-maxlength="@value"}
-  @params.threshold.filter(_ > 0).map {value => data-threshold="@value"}
-  @params.maxWords.filter(_ > 0).map {value => data-maxwords="@value"}
-}
-
 @defining(if(params.language == Cy || messages.lang.code == "cy") Cy else En) { messagesLanguage =>
- <div class="hmrc-character-count" data-module="hmrc-character-count" data-language="@{messagesLanguage.code}"@attrs>
- @govukTextarea(Textarea(
-  id = params.id,
-  name = params.name,
-  describedBy = Some(params.id + "-info"),
-  rows = params.rows,
-  spellcheck = {if (params.spellcheck.isDefined) Some(params.spellcheck.get) else None},
-  value = params.value,
-  formGroupClasses = params.formGroupClasses,
-  classes = "hmrc-js-character-count" + {if (params.errorMessage.isDefined) " govuk-textarea--error" else ""} + {if (params.classes == "") "" else " " + params.classes},
-  label = params.label,
-  hint = params.hint,
-  errorMessage = params.errorMessage,
-  attributes = params.attributes
- ))
-
  @if(messagesLanguage == Cy) {
-  @govukHint(Hint(
-   content = Text("Gallwch nodi hyd at " + params.maxLength.getOrElse(params.maxWords.getOrElse(0)) + {if(params.maxWords.isDefined) " o eiriau" else " o gymeriadau"}),
-   id = Some(params.id + "-info"),
-   classes = "hmrc-character-count__message" + {if (params.countMessageClasses == "") "" else " " + params.countMessageClasses}
-  ))
+  @defining(
+   messages match {
+    case messagesImpl: MessagesImpl => messagesImpl.messagesApi.preferred(Seq(Lang("cy")))
+    case _ => messages
```

Comment:
@JoPintoPaul what do you think about having the welsh translations inline rather than in the messages file and only using the messages implicit to check what the current language and maybe override the passed language param if it seems like they have sent En by mistake? We're adding the english translations to the messages file in this change but not using them

Reply:
@oscarduignan I did dither on this one! In other places we use translated strings inline, I'm not wild about that pattern but maybe it does make sense to be consistent. I'm hoping to go with group consensus on this one.

Reply:
@oscarduignan Yes the English translations are really there as a fallback only but I would struggle to think of a way they can be reached, not sure if the `Messages` would ever fall back to them as default.

Reply:
wish we could ignore params.language and just use the messages implicit with no branching - seems very unlikely that people would want a welsh character count but not have rest of the site using any welsh translations from messages